src/templates/apiReference.js

```diff
@@ -0,0 +1,139 @@
+
+const getMetaDataDetails = (document, prop, alternative = '') =>
+  document && document.meta && document.meta[prop] ? document.meta[prop] : alternative;
+
+const CANONICAL_ROOT = `https://www.ably.com${DOCUMENTATION_PATH}`;
+const META_DESCRIPTION_FALLBACK = `Ably provides a suite of APIs to build, extend, and deliver powerful digital experiences in realtime. Organizations like Toyota, Bloomberg, HubSpot, and Hopin depend on Ably’s platform to offload the growing complexity of business-critical realtime data synchronization at global scale.`;
+const NO_LANGUAGE = 'none';
+
+const ApiReference = ({
+  location: { search },
+  pageContext: { contentOrderedList, languages, version, contentMenu, slug, script },
+  data: { document, versions },
+}) => {
+  const params = new URLSearchParams(search);
+  const language = params.get('lang') ?? DEFAULT_LANGUAGE;
+  useEffect(() => {
+    const preferredLanguage = safeWindow.localStorage.getItem(PREFERRED_LANGUAGE_KEY);
+    if (preferredLanguage && language !== preferredLanguage) {
+      const { isLanguageDefault, isPageLanguageDefault } = getLanguageDefaults(preferredLanguage, language);
+      const href = createLanguageHrefFromDefaults(isPageLanguageDefault, isLanguageDefault, preferredLanguage);
+      navigate(href);
+    }
+  }, []);
+
+  const title = getMetaDataDetails(document, 'title');
+  const description = getMetaDataDetails(document, 'meta_description', META_DESCRIPTION_FALLBACK);
+  const menuLanguages = getMetaDataDetails(document, 'languages', languages);
+  const canonical = `${CANONICAL_ROOT}${slug}`;
+
+  const versionData = {
+    versions: versions.edges,
+    version,
+    rootVersion: slug,
+  };
+
+  const filteredLanguages = useMemo(
+    () =>
+      menuLanguages.includes(NO_LANGUAGE)
+        ? []
+        : menuLanguages
+            .filter((language) => /^[^,]+$/.test(language))
+            .filter((language) => !IGNORED_LANGUAGES.includes(language)),
+    [menuLanguages],
+  );
+  const languagesExist = filteredLanguages.length > 1;
+  const elements = useMemo(
+    () =>
+      contentOrderedList.map(
+        // It is currently safe to use an index as a key.
+        // We will need a unique key if we want to alter any of these by position.
+        ({ data }, i) => <Html data={data} key={i} />,
+      ),
+    [contentOrderedList, language],
+  );
+
+  return (
+    <PageLanguageContext.Provider value={language}>
+      <PageLanguagesContext.Provider value={languages}>
+        <Helmet>
+          <title>{title}</title>
+          <meta property="og:title" content={title} />
+          <meta property="twitter:title" content={title} />
+          <link rel="canonical" href={canonical} />
+          <meta name="description" content={description} />
+          <meta property="og:description" content={description} />
+          <meta name="twitter:description" content={description} />
+        </Helmet>
```

Comment:
Ideally this would be its own component, similarly to Voltaire https://github.com/ably/voltaire/blob/main/src/components/head/head.js, so we don't get discrepancies on what pages can output certain meta tags and which can't